src/community/wmts-multi-dimensional/src/main/java/org/geoserver/gwc/wmts/dimensions/Dimension.java

```diff
@@ -125,11 +150,27 @@ public void addDomainRestriction(Object domainRestriction) {
      * can be NULL.
      */
     public Tuple<ReferencedEnvelope, Tuple<Integer, List<String>>> getDomainValuesAsStrings(Filter filter) {
-        Tuple<ReferencedEnvelope, List<Object>> domainValues = getDomainValues(filter, true);
-        return Tuple.tuple(domainValues.first,
-                Tuple.tuple(domainValues.second.size(), DimensionsUtils.getDomainValuesAsStrings(dimensionInfo, domainValues.second)));
+        if (dimensionInfo.getPresentation() == DimensionPresentation.LIST) {
+            Tuple<ReferencedEnvelope, List<Object>> domainValues = getDomainValues(filter, true);
+            return Tuple.tuple(domainValues.first,
+                    Tuple.tuple(domainValues.second.size(), DimensionsUtils.getDomainValuesAsStrings(dimensionInfo, 
+                            domainValues.second)));
+        } else {
+            // optimize out and get just the min and max values
+            DomainSummary summary = getDomainSummary(filter, false);
+            List<Object> aggregates = new ArrayList<>();
+            if (summary.getMin() != null) {
+                aggregates.add(summary.getMin());
+            }
+            if (summary.getMax() != null && !aggregates.contains(summary.getMax())) {
```

Comment:
This clause ``!aggregates.contains(summary.getMax()`` is here to avoid duplicate values right ? If the min and max are the same we only get one ?

Reply:
That is the intention yes